components/stateful-form.tsx

```diff
@@ -114,16 +114,21 @@ function StatefulForm<Z extends ZodTypeAny>({
 
   const finalSchema = getSchemaForVisibleFields(validationSchema, fields);
 
+  const formConfig: Parameters<typeof useForm<TypeOf<Z>>>[0] = {
```

Comment:
why is it `[0]` , what's that supposed to mean? can you please add comment to make it makes sense a bit? this code looks very cryptic.